packages/flutter_tools/lib/src/ios/fallback_discovery.dart

```diff
@@ -180,6 +176,10 @@ class FallbackDiscovery {
         // No action, we might have failed to connect.
         firstException ??= err;
         _logger.printTrace(err.toString());
+      } on Sentinel {
```

Comment:
Oh, I see, I guess it throws this exception. Is `Sentinel` a subtype of `Exception`?

Reply:
Negative :( @bkonyi !!!!

Reply:
What's still throwing `Sentinel`? Everything should be throwing [SentinelException](https://pub.dev/documentation/vm_service/latest/vm_service/SentinelException-class.html) as of 4.0.0.

Reply:
The changelog in https://pub.dev/packages/vm_service#-changelog-tab- says a sentinel is thrown FYI

Reply:
Ah shoot, must have forgotten to update it. Originally we were going to just throw the Sentinel but then decided to make a special exception for it.

Reply:
Good call :)